Source/Core/Trackers/InspirationTracker.cs

```diff
@@ -8,152 +8,62 @@ namespace PrisonLabor.Core.Trackers
 {
     internal static class InspirationTracker
     {
-        public static Dictionary<Map, Dictionary<Pawn, float>> inspirationValues = new Dictionary<Map, Dictionary<Pawn, float>>();
+        private static Dictionary<Pawn, float> isWatched = new Dictionary<Pawn, float>();
 
-        private static UInt16 tickCounter = 32;
-        private static UInt16 updateInterval = 32;
-
-        /// <summary>
-        /// Check if pawn is watched(supervised) by a Jailor
-        /// </summary>
         public static bool IsWatched(this Pawn pawn)
         {
-            Dictionary<Pawn, float> iValues;
-            float value;
+            isWatched[pawn] = 0;
+
+            var room = pawn.GetRoom();
+
+            if (room == null)
+                return false;
+
+            var wardensCount = 0;
+            var prisonersCount = 0;
 
-            var map = pawn?.Map;
-            lock (inspirationValues)
+            lock (Tracked.LOCK_WARDEN)
             {
-                if (map != null && inspirationValues.TryGetValue(map, out iValues))
-                {
-                    if (iValues.TryGetValue(pawn, out value))
-                    {
-                        return value != 0;
-                    }
-                }
+                wardensCount = Tracked.Wardens[room.ID].Count;
+                prisonersCount = Tracked.Prisoners[room.ID].Count;
             }
-            return false;
-        }
 
-        public static float GetInsiprationValue(Pawn pawn, bool refresh = false)
-        {
-            var map = pawn.Map;
-            if (!inspirationValues.ContainsKey(map) || !inspirationValues[map].ContainsKey(pawn))
+            if (prisonersCount == 0)
             {
-                Log.Message("PrisonLabor Warning: InspirationTracker.GetInsipirationValue() didn't find map or pawn. Maybe it was called before calculating values.");
-                return 0;
+                isWatched[pawn] = 0.0f;
+                return false;
             }
-            lock (inspirationValues)
+
+            if (wardensCount == 0)
             {
-                return inspirationValues[map][pawn];
+                isWatched[pawn] = -0.00425f;
+                return false;
             }
-        }
 
-        public static void Calculate(Map map)
-        {
-            if (InspirationTracker.tickCounter >= InspirationTracker.updateInterval)
+            if (room.IsHuge || room.CellCount > 1600)
```

Comment:
whats the difference?

Reply:
The system now is now event based, and support many types of motivation sources. I'm going to add turrets today (may be in couple of hours)

Reply:
I mean whats the difference between: room.IsHuge || room.CellCount > 1600 :D

Reply:
right lol